ray-operator/controllers/ray/utils/httpproxy_httpclient.go

```diff
@@ -5,30 +5,45 @@ import (
 	"io"
 	"net/http"
 	"time"
+
+	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 type RayHttpProxyClientInterface interface {
 	InitClient()
 	CheckHealth() error
-	SetHostIp(hostIp string, port int)
+	SetHostIp(hostIp, podNamespace, podName string, port int)
 }
 
-func GetRayHttpProxyClient() RayHttpProxyClientInterface {
-	return &RayHttpProxyClient{}
+func GetRayHttpProxyClientFunc(mgr ctrl.Manager, useProxy bool) func() RayHttpProxyClientInterface {
+	return func() RayHttpProxyClientInterface {
+		return &RayHttpProxyClient{
+			mgr:      mgr,
+			useProxy: useProxy,
+		}
+	}
 }
 
 type RayHttpProxyClient struct {
-	client       http.Client
+	client       *http.Client
 	httpProxyURL string
+
+	mgr      ctrl.Manager
+	useProxy bool
```

Comment:
Nit: It's better to use `UseKubernetesProxy` for consistency.

No need to update this in this PR. I recently worked on an open-source event for newbies. This would be a good first issue for the event. I will open an issue later.